src/MainUtils.cpp

```diff
@@ -448,10 +448,12 @@ void addONNXToKrnlPasses(mlir::PassManager &pm) {
   pm.addNestedPass<FuncOp>(createDisconnectKrnlDimFromAllocPass());
 
   // TODO: make this pass optional:
-  pm.addNestedPass<FuncOp>(mlir::createKrnlEnableMemoryPoolPass());
-  pm.addNestedPass<FuncOp>(mlir::createKrnlBundleMemoryPoolsPass());
-  pm.addPass(mlir::createCanonicalizerPass());
-  pm.addNestedPass<FuncOp>(mlir::createKrnlOptimizeMemoryPoolsPass());
+  // These passes causes error when compling LSTM/GRU/RNN ops with unknown dims.
+  // Temporary disable them for testing purpose.
+  // pm.addNestedPass<FuncOp>(mlir::createKrnlEnableMemoryPoolPass());
+  // pm.addNestedPass<FuncOp>(mlir::createKrnlBundleMemoryPoolsPass());
+  // pm.addPass(mlir::createCanonicalizerPass());
+  // pm.addNestedPass<FuncOp>(mlir::createKrnlOptimizeMemoryPoolsPass());
   pm.addPass(mlir::createCanonicalizerPass());
```

Comment:
Will these changes go into the patch or do not run those test cases until the source of error is fixed?

Reply:
I narrowed down the issue to the BundleMemoryPools pass and reported here #431. I am not sure which one to go. It seems to me there is an issue in BundleMemoryPools pass rather than in LSTM lowering code, since LSTM tests passed when the pass is turned off.

Reply:
I suggest you leave the LSTM out of the dynamic test currently but do not change the intended passes.

Reply:
OK. I updated according to your comments. Thanks!